velox/vector/VectorStream.cpp

```diff
@@ -17,29 +17,33 @@
 #include <memory>
 
 namespace facebook::velox {
-
 namespace {
-std::unique_ptr<VectorSerde>& getVectorSerde() {
+std::unique_ptr<VectorSerde>& getVectorSerdeImpl() {
   static std::unique_ptr<VectorSerde> serde;
   return serde;
 }
 } // namespace
 
-bool registerVectorSerde(std::unique_ptr<VectorSerde> serde) {
-  VELOX_CHECK(!getVectorSerde().get(), "Vector serde is already registered");
-  getVectorSerde() = std::move(serde);
-  return true;
+VectorSerde* getVectorSerde() {
+  auto serde = getVectorSerdeImpl().get();
+  VELOX_CHECK_NOT_NULL(serde, "Vector serde is not registered");
+  return serde;
+}
+
+void registerVectorSerde(std::unique_ptr<VectorSerde> serdeToRegister) {
```

Comment:
It seems that registerVectorSerde are not thread-safe? Either using a static lock inside to protect the setting or explicitly comment in header say we don't expect concurrent call to this function? Thanks!

Reply:
Yeah since this call is only designed to be made once in the system, probably not worth a lock. I'll add comments! Thanks!